lib/hexpm/repository/package_reports.ex

```diff
@@ -32,25 +40,110 @@ defmodule Hexpm.Repository.PackageReports do
     |> Repo.one()
     |> PackageReport.change_state(%{"state" => "accepted"})
     |> Repo.update()
+
+    report = PackageReport.get(report_id)
+
+    report =
+      PackageReport.get(report_id)
+      |> Repo.one()
+
+    email_user_about_state_change(report, report.author)
+
+    Enum.each(
+      Owners.all(report.package),
+      fn owner ->
+        owner = Hexpm.Repo.preload(owner, user: [])
```

Comment:
You can preload on the call to Owners.all instead. This avoids the N+1 query problem.

https://stackoverflow.com/questions/97197/what-is-the-n1-selects-problem-in-orm-object-relational-mapping